File: cpp/operating-system-specific-dependencies/app/src/main/cpp/logger.cpp
/*
 * A logger that writes messages to a console.
 */
#include <iostream>

#ifdef _WIN32
#include "win_console.h"
#else
#include "ansi_console.h"
#endif
#include "logger.h"

logger::logger() {
}

void logger::info(const std::string & message) {
#ifdef _WIN32
    win_console console;
#else
    ansi_console console;
#endif
    console.color(message);
}

File: swift/operating-system-specific-dependencies/app/src/main/swift/Logger.swift
/*
 * A logger that writes to the console.
 */
#if os(Linux)
    import LinuxConsole
#else
    import MacOsConsole
#endif

class Logger {
    let console = Console()

    // Writes an info message and trailing new-line
    func info(_ s: String) {
        console.color(s)
    }
}
